src/pool/cli.c

```diff
@@ -101,11 +102,10 @@ dc_pool_init(void)
 	}
 
 	pproto->ep.ep_grp = sys->sy_group;
-	rc = rsvc_client_choose(&pproto->cli, &pproto->ep);
-	if (rc) {
-		D_ERROR("rsvc_client_choose() failed: "DF_RC"\n", DP_RC(rc));
-		D_GOTO(out_rsvc, rc);
-	}
+	pproto->ep.ep_tag = 0;
+
+	num_ranks = dc_mgmt_net_get_num_srv_ranks();
+	pproto->ep.ep_rank = rand() % num_ranks;
```

Comment:
The randomly chosen rank may be down when the client is run. This could result (in one example I tried) in -DER_HG. And in query_cb() the retryable RPC logic will then select a rank from among sys->sy_info.ms_ranks. Is that the intention, to try wide scalability first (from among all engine ranks in the system) and if it fails, choose from a smaller set of ranks that (perhaps?) are more likely to be up/available?

Reply:
good point, i think there are same question for query_cb() in src/object/cli_mod.c

I guess original PR https://github.com/daos-stack/daos/commit/db999ddec5b3b98364d31111aef7520331de6ce1 did not
take this account as well, so if we do rand() retry ranks, we need remember what we have tried, implement similar logic like what rsvc_client_choose() did, but maybe still rand to select next one.

If we consider server down not a frequent case, retry with smaller ranks might be enough to avoid performance issue for most of time i guess, notice if all those small ranks failed as well, even we could try more ranks, but pool query/or system will stop working as well.

Reply:
For now those are stop-gap measures to prevent clients hitting a single rank for protocol query until https://daosio.atlassian.net/browse/DAOS-10883 is implemented.